src/lib/layouts/workflow-history-layout.svelte

```diff
@@ -22,6 +22,11 @@
 
   const { namespace } = $page.params;
   const { workflow, workers } = $workflowRun;
+  const namespaces = $page.stuff?.namespaces;
+
+  $: parentNamespace = namespaces?.find(
```

Comment:
I'd break this out and wrap it in a unit test. If you count every `?.` as a conditional, there are lots of fun ways things could go wrong.